src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataCollectionSerializer.cs

```diff
@@ -161,11 +174,22 @@ public ODataCollectionSerializer(ODataSerializerProvider serializerProvider)
                     IEdmTypeReference actualType = writeContext.GetEdmType(item, item.GetType());
                     Contract.Assert(actualType != null);
 
-                    itemSerializer = itemSerializer ?? SerializerProvider.GetEdmTypeSerializer(actualType);
                     if (itemSerializer == null)
                     {
-                        throw new SerializationException(
-                            Error.Format(SRResources.TypeCannotBeSerialized, actualType.FullName()));
+                        //For instance annotations we need to use complex serializer for complex type and not ODataResourceSerializer
+                        if (isForAnnotations && actualType.IsComplex())
```

Comment:
>IsComplex [](start = 59, length = 9)

to sam's point above, now that we've changed ODataComplexValueSerializer to ODataReosurceValueSerializer, should we be checking actualType.IsComplex() or actualType.IsStructured here?